src/de/deserializer.zig

```diff
@@ -66,150 +67,137 @@ pub fn Deserializer(comptime user_dbt: anytype) type {
             const Visitor = @TypeOf(visitor);
             const visitor_info = @typeInfo(Visitor);
 
-            const tokens = self.tokens;
-
-            if (try self.tokens.next()) |token| {
-                switch (token) {
-                    .True, .False => {
-                        return try visitor.visitBool(allocator, De, token == .True);
-                    },
-                    .Number => |t| {
-                        const slice = t.slice(self.tokens.slice, self.tokens.i - 1);
-
-                        // Enum
-                        if (visitor_info == .Enum) {
-                            if (t.is_integer) {
-                                switch (slice[0]) {
-                                    '0'...'9' => return try visitor.visitInt(allocator, De, try std.fmt.parseInt(u128, slice, 10)),
-                                    else => return try visitor.visitInt(allocator, De, try std.fmt.parseInt(i128, slice, 10)),
-                                }
+            const token = try self.tokens.nextAlloc(allocator orelse return error.MissingAllocator, .alloc_if_needed);
+            defer self.freeToken(token);
+
+            switch (token) {
+                .true, .false => {
+                    return try visitor.visitBool(allocator, De, token == .true);
+                },
+                inline .number, .allocated_number => |slice| {
+                    const is_integer = std.json.isNumberFormattedLikeAnInteger(slice);
```

Comment:
nit

We can rearrange this section so that we only call `isNumberFormattedLikeAnInteger` when we actually need it.

To do so, we can move the `if (visitor_info == .Int)` block up to the top of this branch. Then, the `if (visitor_info == .Enum)` block will be below. Then, we have the float section with `if (std.json.isNumberFormattedLikeAnInteger)`. And then after that the `return switch(slice[0])` for integers with no type hints.